### PR TITLE
fix copycds can't auto copycd centos iso

### LINKED ISSUE #2303: copycds with centos 1611 iso image with wrong template call
OS :CentOS 1611 + xCAT 2.13.0
 
copycds -n centos7.3 CentOS-7-x86_64-DVD-1611.iso   did not build a correct linuximage object 

[root@mgt ~]# lsdef -t osimage -o centos7.3-x86_64-install-compute
Object name: centos7.3-x86_64-install-compute
    imagetype=linux
    osarch=x86_64
    osdistroname=centos7.3-x86_64
    osname=Linux
    osvers=centos7.3
    otherpkgdir=/install/post/otherpkgs/centos7.3/x86_64
    pkgdir=/install/centos7.3/x86_64
   pkglist=/opt/xcat/share/xcat/install/centos/compute.pkglist
    profile=compute
    provmethod=install
    template=/opt/xcat/share/xcat/install/centos/compute.tmpl

and this issue can be fixed with following command 

 213  chdef -t osimage -o centos7.3-x86_64-install-compute  template=/opt/xcat/share/xcat/install/centos/compute.centos7.tmpl
  214  chdef -t osimage -o centos7.3-x86_64-install-compute  pkglist=/opt/xcat/share/xcat/install/centos/compute.centos7.pkglist

### PR DESCRIPTION
**Pull request by penguhyang**
fix #2303

### CODE REVIEW

**Comment by immarvin** [comment]
the .discinfo and .treeinfo in centos7.2 iso:
````
[root@c910f03c05k21 xcat-core]# mount /iso/iso/centos/7.2/ppc64le/CentOS-7-AltArch-ppc64le-Everything-1511.iso /mnt/
mount: /dev/loop3 is write-protected, mounting read-only
[root@c910f03c05k21 xcat-core]# cat /mnt/
boot/                                     LiveOS/                                   RPM-GPG-KEY-CentOS-SIG-AltArch-7-ppc64le
.discinfo                                 Packages/                                 TRANS.TBL
EULA                                      ppc/                                      .treeinfo
GPL                                       repodata/
images/                                   RPM-GPG-KEY-CentOS-7
[root@c910f03c05k21 xcat-core]# cat /mnt/.discinfo
1450147276.351714
151215
ppc64le
[root@c910f03c05k21 xcat-core]# cat /mnt/.treeinfo
[general]
name = CentOS AltArch-7
family = CentOS AltArch
timestamp = 1450148160.41
variant =
version = 7
packagedir =
arch = ppc64le

[stage2]
mainimage = LiveOS/squashfs.img

[images-ppc64le]
kernel = ppc/ppc64/vmlinuz
initrd = ppc/ppc64/initrd.img
boot.iso = images/boot.iso
zimage = images/netboot/ppc64.img
[root@c910f03c05k21 xcat-core]# cat /mnt/.discinfo
1450147276.351714
151215
ppc64le
````

can your code recognize it?

File: xCAT-server/lib/xcat/plugins/anaconda.pm [hunk 1]
```diff
@@ -2237,6 +2237,18 @@ sub copycd
         }
     }
 
+    open($dinfo, $mntpath . "/.treeinfo");
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/anaconda.pm]: this section should be placed in the correct place in the loop in line 2252

File: xCAT-server/lib/xcat/plugins/anaconda.pm [hunk 2]
```diff
@@ -2268,8 +2268,22 @@ sub copycd
         }
         else
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/anaconda.pm]: add a new branch ``elsif ($desc =~ /^[\d\.]+$/)`` and include your code logic for centos, keep the original else branch

Comment (penguhyang) [inline xCAT-server/lib/xcat/plugins/anaconda.pm]: fixed.

File: xCAT-server/lib/xcat/plugins/anaconda.pm [hunk 3]
```diff
@@ -2266,6 +2266,20 @@ sub copycd
...
             close($dinfo);
         }
+        elsif ($desc =~ /^[\d\.]+$/)
+        {
+            open($dinfo, $mntpath . "/.treeinfo");
+            while (<$dinfo>) {
+                chomp($_);
+                s/\s+$//;    #remove trailing spaces
+                next if /^\s*$/;    #-- skip empty lines
+                if ($_ =~ /family = CentOS/) {
```
Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/anaconda.pm]: should better be ``if ($_ =~ /family\s*=\s*CentOS/i) {``

**Comment by immarvin** [comment]
agree to merge

**Comment by daniceexi** [comment]
Thanks for the change. 
I prefer to have more comments for what's this change for so that other programer can understand this part of code easily.

**Pull request merged**